pkg/api/transform_network.go

```diff
@@ -63,23 +63,50 @@ func TransformNetworkOperationName(operation string) string {
 }
 
 type NetworkTransformRule struct {
-	Input           string        `yaml:"input,omitempty" json:"input,omitempty" doc:"entry input field"`
-	Output          string        `yaml:"output,omitempty" json:"output,omitempty" doc:"entry output field"`
-	Type            string        `yaml:"type,omitempty" json:"type,omitempty" enum:"TransformNetworkOperationEnum" doc:"one of the following:"`
-	Parameters      string        `yaml:"parameters,omitempty" json:"parameters,omitempty" doc:"parameters specific to type"`
-	Assignee        string        `yaml:"assignee,omitempty" json:"assignee,omitempty" doc:"value needs to assign to output field"`
-	KubernetesInfra *K8sInfraRule `yaml:"kubernetes_infra,omitempty" json:"kubernetes_infra,omitempty" doc:"Kubernetes infra rule specific configuration"`
-	Kubernetes      *K8sRule      `yaml:"kubernetes,omitempty" json:"kubernetes,omitempty" doc:"Kubernetes rule specific configuration"`
+	Type            string                 `yaml:"type,omitempty" json:"type,omitempty" enum:"TransformNetworkOperationEnum" doc:"one of the following:"`
+	KubernetesInfra *K8sInfraRule          `yaml:"kubernetes_infra,omitempty" json:"kubernetes_infra,omitempty" doc:"Kubernetes infra rule configuration"`
+	Kubernetes      *K8sRule               `yaml:"kubernetes,omitempty" json:"kubernetes,omitempty" doc:"Kubernetes rule configuration"`
+	AddSubnet       *NetworkAddSubnetRule  `yaml:"add_subnet,omitempty" json:"add_subnet,omitempty" doc:"Add subnet rule configuration"`
+	AddLocation     *NetworkGenericRule    `yaml:"add_location,omitempty" json:"add_location,omitempty" doc:"Add location rule configuration"`
+	AddIPCategory   *NetworkGenericRule    `yaml:"add_ip_category,omitempty" json:"add_ip_category,omitempty" doc:"Add ip category rule configuration"`
+	AddService      *NetworkAddServiceRule `yaml:"add_service,omitempty" json:"add_service,omitempty" doc:"Add service rule configuration"`
```

Comment:
It seems we often have same type of rules applied in a row; on both source & destination for example.

Would it make sense to allow arrays on the objects above ?

Then you will be able to call something like:
```
api.NetworkTransformRules{{
    Type: api.AddKubernetesRuleType,
    Kubernetes: []api.K8sRule{
      api.K8sRule{
          Input:  "SrcAddr",
          Output: "SrcK8S",
      }, 
      api.K8sRule{
          Input:  "DstAddr",
          Output: "DstK8S",
      },
    },
}}})
```

We could even get rid of the type by just providing objects :thinking:

Reply:
We could remove the type object but this would change consistency with the rest of the API that is using it.

About allowing an array, this mean we would have two nested level of rules instead of having one flat array of rules. This add complexity and I am not sure to see a value in doing it.